### PR TITLE
1134 fix isolines geom

### PR DESCRIPTION
**Pull request by Jesus89**
Fixes https://github.com/CartoDB/cartoframes/issues/1134

- Add `with_lnglat` param to Isolines services
- Refactor `geodataframe_from_dataframe` method
- Add/Update tests
- Update examples

### CODE REVIEW

File: cartoframes/data/services/isolines.py [hunk 1]
```diff
@@ -146,6 +146,7 @@ def _iso_areas(self,
         else:  # source.is_local()
             # upload to temporary table
             temporary_table_name = self._new_temporary_table_name()
+            compute_geodataframe(source)
```
Comment (jgoizueta) [inline cartoframes/data/services/isolines.py]: Couldn't we rely on `upload` converting lng/lat to geometry on the fly? I recall there's a `with_lnglat` option in the method that generates CSV data to be copied but I don't remember if that works automatically.

Comment (Jesus89) [reply cartoframes/data/services/isolines.py]: There is no automatic detection for the upload, because you need to pass explicitly the parameters.

**Comment by Jesus89** [comment]
Yes. You're right, we should not modify the DF/GDF of the user. To improve that, there are two solutions:
- Copy the dataframe internally, which means duplicate the mem usage (not good)
- Add a temporary column to the dataframe for the geometry with a reserved name and then remove that column after the operation (better)

**Comment by Jesus89** [comment]
Hi @jgoizueta.

I finally implemented a different solution, which is passing the `with_lnglat` parameter to the `isolines` and `isochrones` services. So it works the same as the upload method. I have updated the tests and also the examples adding this parameter.

As a bonus, I have improved the `geodataframe_from_dataframe` method to use a reserved-word column for the geometry and some minor fixes to improve the compatibility with geopandas.

Could you take a second review? Thanks!

**Pull request merged**